### PR TITLE
Added language and misspell check for markdown files

### LINKED ISSUE #407: Create a Markdown spellcheck Github Action
There is the possibility to check spell errors in Markdown and other kind of document files. It would be nice if we could add a Github Action to do so in clevis. Examples on how this can be done can be checked here:
https://www.javierbrea.com/blog/spellcheck-github-action/
https://github.com/sarroutbi/rustspyder/commit/1aec74a656a0770df66aae8ba3687dd09dcfb37e
https://github.com/sarroutbi/koji-retriever/pull/87/commits/4989c2a455139c2e88b7f29aafc6424609c2536a

### PR DESCRIPTION
**Pull request by yo-404**
### Description

Changes proposed in this pull request:

- Added Language Tool
- Added Misspell check

Files Added

    spellcheck.yaml @.github\workflows\

### results
image

image

image

currently level is set to `warning` so that it doesnt stops the PR and CI will succeed with info shown as warning . Can be changed to `error` if required

### CODE REVIEW

File: .github/workflows/spellcheck.yml [hunk 1]
```diff
@@ -0,0 +1,42 @@
+name: Spellcheck
+on: 
+  push:
```
Comment (sarroutbi) [inline .github/workflows/spellcheck.yml:3]: Could you please include it for pull requests?:

```
on:
  push:
    branches: ["main"]
  pull_request:
```

**Review by sarroutbi — CHANGES_REQUESTED** [review]

File: .github/workflows/spellcheck.yml (hunk 1 above)
Comment (yo-404) [reply .github/workflows/spellcheck.yml:3]: sure

**Review by yo-404 — COMMENTED** [review]
Added pull_request

**Review by sarroutbi — APPROVED** [review]
Change LGTM

**Pull request merged**